group-coordinator/src/main/java/org/apache/kafka/coordinator/group/modern/consumer/ConsumerGroup.java

```diff
@@ -514,6 +515,13 @@ public void validateOffsetCommit(
         // the request can commit offsets if the group is empty.
         if (memberEpoch < 0 && members().isEmpty()) return;
 
+        // TxnOffsetCommitRequest versions v0-v2 do not include a member ID.
+        // And we can still send UNKNOWN_GENERATION_ID and UNKNOWN_MEMBER_ID through Producer#sendOffsetsToTransaction.
+        // Therefore, we should not throw exception in these cases.
```

Comment:
I would simplify it a bit. What about `The TxnOffsetCommit API does not require the member id, the generation id and the group instance id fields. Hence the are only validated if any of them is provided.`?